presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentFileUtils.java

```diff
@@ -42,12 +43,12 @@
     private AgentFileUtils() {}
 
     public static List<String> getFileNameList(Path path)
+            throws IOException
     {
-        File folder = path.toFile();
-        checkArgument(folder.isDirectory(), "%s is not a directory", path);
+        checkArgument(path.toFile().isDirectory(), "%s is not a directory", path);
         ImmutableList.Builder<String> fileNames = ImmutableList.builder();
-        for (int i = 0; i < folder.list().length; i++) {
-            fileNames.add(folder.list()[i]);
+        try (Stream<Path> stream = Files.list(path)) {
+            stream.filter(Files::isRegularFile).forEach(p -> fileNames.add(p.getFileName().toString()));
         }
```

Comment:
To collect a stream into an immutable list, either of these is a nicer strategy:
```java
stream.collect(collectingAndThen(Collectors.toList(), ImmutableList::copyOf));
ImmutableList.copyOf(stream.collect(Collectors.toList()));
```